<a href="https://colab.research.google.com/github/Faabry/Pandas-Repository/blob/main/Pandas_An%C3%A1lise_de_vendas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Importando uma tabela a partir de um arquivo Excel**

In [2]:
import pandas as pd

df = pd.read_excel('Vendas.xlsx')
df

Código Venda       Data              ID Loja            Produto  \
0                 1 2019-01-01   Iguatemi Esplanada     Sapato Estampa   
1                 1 2019-01-01   Iguatemi Esplanada           Camiseta   
2                 1 2019-01-01   Iguatemi Esplanada      Sapato Xadrez   
3                 2 2019-01-02       Norte Shopping            Relógio   
4                 2 2019-01-02       Norte Shopping       Chinelo Liso   
...             ...        ...                  ...                ...   
93905         65012 2019-11-30  Shopping Vila Velha     Mochila Xadrez   
93906         65013 2019-11-30    Ribeirão Shopping  Pulseira Listrado   
93907         65013 2019-11-30    Ribeirão Shopping     Cueca Listrado   
93908         65014 2019-11-30     Shopping Morumbi     Pulseira Linho   
93909         65014 2019-11-30     Shopping Morumbi      Casaco Xadrez   

       Quantidade  Valor Unitário  Valor Final  
0               1             358          358  
1               2             180          360  
2               1             368          368  
3               3             200          600  
4               1              71           71  
...           ...             ...          ...  
93905           2             283          566  
93906           2              79          158  
93907           3              67          201  
93908           5             114          570  
93909           4             259         1036  

[93910 rows x 7 columns]

# **Qual foi a loja com o maior faturamento?**


In [113]:
# O primeiro passo é agrupar todas as lojas para que possamos calcular
# o faturamento de cada loja individualmente
lojas = df.groupby('ID Loja').sum()
lojas

Código Venda  Quantidade  Valor Unitário  \
ID Loja                                                                      
Bourbon Shopping SP                  123249345        8236          736434   
Center Shopping Uberlândia           118172000        7994          715956   
Iguatemi Campinas                    124216893        8228          731584   
Iguatemi Esplanada                   121435795        8008          729290   
Norte Shopping                       128314645        8468          749628   
Novo Shopping Ribeirão Preto         124275928        8234          725159   
Palladium Shopping Curitiba          125200233        8446          738810   
Parque Dom Pedro Shopping            117175394        7789          702186   
Passei das Águas Shopping            119631947        7999          715348   
Ribeirão Shopping                    121531534        8222          732663   
Rio Mar Recife                       124333829        8096          729370   
Rio Mar Shopping Fortaleza           121529774        8308          728484   
Salvador Shopping                    122417527        8029          707881   
Shopping Barra                       120308443        8067          718236   
Shopping Center Interlagos           122665281        8130          722366   
Shopping Center Leste Aricanduva     126494842        8447          732056   
Shopping Eldorado                    120097450        8075          703116   
Shopping Ibirapuera                  121325317        8169          708822   
Shopping Iguatemi Fortaleza          120395392        7989          718861   
Shopping Midway Mall                 114715773        7659          688577   
Shopping Morumbi                     116391208        7855          690648   
Shopping Recife                      115854324        7929          686817   
Shopping SP Market                   125397773        8343          733653   
Shopping União de Osasco             121336469        8122          712678   
Shopping Vila Velha                  132298698        8619          746355   

                                  Valor Final  
ID Loja                                        
Bourbon Shopping SP                   1600260  
Center Shopping Uberlândia            1543961  
Iguatemi Campinas                     1612460  
Iguatemi Esplanada                    1596339  
Norte Shopping                        1610477  
Novo Shopping Ribeirão Preto          1578522  
Palladium Shopping Curitiba           1600499  
Parque Dom Pedro Shopping             1509575  
Passei das Águas Shopping             1536925  
Ribeirão Shopping                     1594186  
Rio Mar Recife                        1582897  
Rio Mar Shopping Fortaleza            1570540  
Salvador Shopping                     1517583  
Shopping Barra                        1549955  
Shopping Center Interlagos            1536665  
Shopping Center Leste Aricanduva      1596714  
Shopping Eldorado                     1523772  
Shopping Ibirapuera                   1527153  
Shopping Iguatemi Fortaleza           1550642  
Shopping Midway Mall                  1485172  
Shopping Morumbi                      1468508  
Shopping Recife                       1493008  
Shopping SP Market                    1610930  
Shopping União de Osasco              1547738  
Shopping Vila Velha                   1615271

In [117]:
# Agora podemos calcular qual foi a loja com o maior faturamento
lojas['Valor Final'].max()
lojas.loc[lojas['Valor Final'] == 1615271]

# A loja com o maior faturamento foi "Shopping Vila Velha" com o faturamento de R$1.615.271,00

Código Venda  Quantidade  Valor Unitário  Valor Final
ID Loja                                                                   
Shopping Vila Velha     132298698        8619          746355      1615271

# **Qual foi a loja com o menor faturamento?**

In [119]:
lojas['Valor Final'].min()
lojas.loc[lojas['Valor Final'] == 1468508]

# A Loja com o menor faturamento foi "Shopping Morumbi" com o faturamento de R$1.468.508,00

Código Venda  Quantidade  Valor Unitário  Valor Final
ID Loja                                                                
Shopping Morumbi     116391208        7855          690648      1468508

In [130]:
maior_faturamento = 1615271
menor_faturamento = 1468508

dif_faturamento = maior_faturamento - menor_faturamento
print(f'R${float(dif_faturamento):.2f}')

# Um diferença de R$146.763,00 entre o maior e o menor faturamento

R$146763.00


In [132]:
# Agora sim podemos ter uma visão da média de faturamento de todas as lojas
lojas['Valor Final'].mean()

# A média de faturamento é R$1.558.390,08

1558390.08

# **Agora vamos descobrir qual foi o produto mais vendido**

In [81]:
# Primeiro precisamos agrupar os produtos para verificar a maior quantidade vendida
# em uma só transação
produtos = df.groupby('Produto').sum()
produtos

Código Venda  Quantidade  Valor Unitário  Valor Final
Produto                                                                
Bermuda               25770079        1692          116100       253800
Bermuda Estampa       24703185        1616          127257       273104
Bermuda Linho         28670449        1893          167115       369135
Bermuda Liso          25053304        1637          120262       257009
Bermuda Listrado      25576302        1675          129259       273025
...                        ...         ...             ...          ...
Tênis Estampa         25479194        1663          200704       425728
Tênis Linho           25535255        1713          232848       503622
Tênis Liso            26076935        1656          205086       440496
Tênis Listrado        25533339        1737          207766       455094
Tênis Xadrez          26118680        1661          205905       430199

[120 rows x 4 columns]

In [79]:
# Agora precisamos saber qual foi a maior quantidade vendida
produtos['Quantidade'].max()

# A maior quantidade de vendas de um unico produto foi de 1893

1893

In [82]:
# Agora precisamos saber qual foi o produto mais vendido
produtos.loc[produtos['Quantidade'] == 1893]

# O produto mais vendido foi a "bermuda linho" com 1893 quantidades vendidas

Código Venda  Quantidade  Valor Unitário  Valor Final
Produto                                                             
Bermuda Linho      28670449        1893          167115       369135

# **Agora vamos descobrir qual foi o produto menos vendido**

In [84]:
produtos['Quantidade'].min()

# A menor quantidade de vendas de um unico produto foi 1529

1529

In [85]:
# Vamos descobrir qual foi o produto menos vendido
produtos.loc[produtos['Quantidade'] == 1529]

# O produto menos vendido foi "short listrado" com 1529 quantidades vendidas

Código Venda  Quantidade  Valor Unitário  Valor Final
Produto                                                              
Short Listrado      22766343        1529           73134       155958

# **Agora vamos quais foram os produtos que tiveram o maior e o menor faturamento**

In [86]:
produtos['Valor Final'].max()
# O produto com maior faturamento foi 1.215.935,00

1215935

In [87]:
# Qual foi o produto mais vendido ?
produtos.loc[produtos['Valor Final'] == 1215935]

# O produto com maior faturamento foi "terno listrado"

Código Venda  Quantidade  Valor Unitário  Valor Final
Produto                                                              
Terno Listrado      24639302        1715          546639      1215935

In [89]:
# Qual foi o produto com menor faturamento?
produtos['Valor Final'].min()

# O produto com menor faturamento teve R$51.450,00

51450

In [90]:
# Qual foi o produto ?
produtos.loc[produtos['Valor Final'] == 51450]

# O produto com menor faturamento foi "meia"

Código Venda  Quantidade  Valor Unitário  Valor Final
Produto                                                       
Meia         26623385        1715           23490        51450

# **Tanto o produto com maior faturamento quanto o produto com menor faturamento foram vendidas exatas 1715 unidades**

# **Agora vamos ver qual é o produto mais caro e o mais barato**

In [102]:
# Primeiro precisamos saber qual foi o maior valor unitário, lembrando que:
# Como estamos usando agrupamento dos produtos o valor retornado sera a soma
# de todos os valores individuais daquele mesmo produto, será necessário depois
# descobrir o valor individual daquele produto
produtos['Valor Unitário'].max()

573120

In [99]:
produtos.loc[produtos['Valor Unitário'] == 573120]

# O produto com o maior valor unitário é "Terno Liso"
# Agora precisamos saber o seu valor individual

Código Venda  Quantidade  Valor Unitário  Valor Final
Produto                                                          
Terno Liso      26118913        1666          573120      1199520

In [100]:
df.loc[df['Produto'] == 'Terno Liso']

# O valor unitário do produto "Terno Liso" é de R$720,00

Código Venda       Data                     ID Loja     Produto  \
215             154 2019-01-02           Shopping Eldorado  Terno Liso   
246             178 2019-01-02            Shopping Morumbi  Terno Liso   
348             251 2019-01-03   Passei das Águas Shopping  Terno Liso   
606             412 2019-01-04            Shopping Morumbi  Terno Liso   
674             454 2019-01-04           Ribeirão Shopping  Terno Liso   
...             ...        ...                         ...         ...   
93721         64891 2019-11-30          Shopping SP Market  Terno Liso   
93819         64953 2019-11-30  Rio Mar Shopping Fortaleza  Terno Liso   
93827         64960 2019-11-30           Iguatemi Campinas  Terno Liso   
93837         64970 2019-11-30           Iguatemi Campinas  Terno Liso   
93855         64982 2019-11-30           Salvador Shopping  Terno Liso   

       Quantidade  Valor Unitário  Valor Final  
215             1             720          720  
246             2             720         1440  
348             2             720         1440  
606             1             720          720  
674             1             720          720  
...           ...             ...          ...  
93721           1             720          720  
93819           3             720         2160  
93827           1             720          720  
93837           1             720          720  
93855           1             720          720  

[796 rows x 7 columns]

In [104]:
# Agora vamos descobrir o produto mais barato
produtos['Valor Unitário'].min()

# A menor quantidade na coluna "valor unitário foi" 23490

23490

In [107]:
# Vamos ver qual produto tem esse valor unitário
produtos.loc[produtos['Valor Unitário'] == 23490]

# O produto com o menor valor unitário é "meia", vamos descobrir de fato qual é 
# seu valor unitário

Código Venda  Quantidade  Valor Unitário  Valor Final
Produto                                                       
Meia         26623385        1715           23490        51450

In [109]:
df.loc[df['Produto'] == 'Meia']
# Como podemos observar o valor unitário do produto "meia" é de R$30,00

Código Venda       Data                       ID Loja Produto  \
27               22 2019-01-02     Parque Dom Pedro Shopping    Meia   
808             541 2019-01-04          Shopping Midway Mall    Meia   
1094            726 2019-01-05   Shopping Iguatemi Fortaleza    Meia   
1599           1059 2019-01-07  Novo Shopping Ribeirão Preto    Meia   
1690           1121 2019-01-07                Shopping Barra    Meia   
...             ...        ...                           ...     ...   
93424         64673 2019-11-29             Salvador Shopping    Meia   
93440         64686 2019-11-29     Passei das Águas Shopping    Meia   
93567         64776 2019-11-29                Rio Mar Recife    Meia   
93745         64908 2019-11-30            Iguatemi Esplanada    Meia   
93804         64945 2019-11-30                Shopping Barra    Meia   

       Quantidade  Valor Unitário  Valor Final  
27              1              30           30  
808             5              30          150  
1094            1              30           30  
1599            4              30          120  
1690            3              30           90  
...           ...             ...          ...  
93424           2              30           60  
93440           3              30           90  
93567           3              30           90  
93745           4              30          120  
93804           1              30           30  

[783 rows x 7 columns]

# **Além de ser o produto mais barato, o produto "Meia" também foi o produto menos vendido**